### PR TITLE
u-boot-at91_2020.01.bb: Backport a patch to fix build on newer hosts

### PR DESCRIPTION
**Pull request by kraj**
hosts with gcc10+ fail to compile u-boot older than 2020.04, a backport
is needed, submitting that thusly

Signed-off-by: Khem Raj <raj.khem@gmail.com>

### CODE REVIEW

**Comment by ehristev** [comment]
I cherry-picked the original commit from u-boot/master to u-boot-at91/u-boot-2020.01-at91 branch, such that it will build with gcc 10 when we update the SHA for u-boot recipe.
Is this fine with you ? We want to avoid having patches for components inside meta-atmel layers.

**Comment by kraj** [comment]
thats fine too. let me know when thats ready in master

**Comment by codrin989** [comment]
I see that this commit https://github.com/linux4sam/meta-atmel/pull/185/commits/6865c048e9b50d7be57989ba50d0463fdd80f2f4 went in a separate pull request and I think it's better this way. Could you please remove it from this pull request?

File: recipes-bsp/dt-overlay-at91/dt-overlay-at91_git.bb [hunk 1]
```diff
@@ -24,6 +24,12 @@ AT91BOOTSTRAP_MACHINE ??= "${MACHINE}"
 do_compile[depends] += "virtual/kernel:do_deploy virtual/kernel:do_shared_workdir"
 do_compile[nostamp] = "1"
 
+do_configure_append() {
+    if ${@bb.utils.contains('INITRAMFS_IMAGE_BUNDLE','1','true','false',d)}; then
+        sed -i -e "s#\"./zImage\"#\"./zImage-initramfs-${MACHINE}.bin\"#" ${S}/${AT91BOOTSTRAP_MACHINE}.its
```
Comment (noglitch) [inline recipes-bsp/dt-overlay-at91/dt-overlay-at91_git.bb:29]: Hi,
Nit: isn't it possible to use python built-in capabilities to do the same?

Comment (kraj) [reply recipes-bsp/dt-overlay-at91/dt-overlay-at91_git.bb:29]: @noglitch I am not sure if I got your question, here we need to evaluate a bitbake variable and edit the source code based on result. full python capabilities are used e.g for assigning values to variables but not when operation require there evaluations. secondly,  do_configure is a shell function, having a function/task fully written in python may give opportunities to write it in python too

**Comment by kraj** [comment]
> I see that this commit [6865c04](https://github.com/linux4sam/meta-atmel/commit/6865c048e9b50d7be57989ba50d0463fdd80f2f4) went in a separate pull request and I think it's better this way. Could you please remove it from this pull request?

ah, the pull came from `mut` branch which keeps changing as I do changes so this one can be closed as I have opened a new one here https://github.com/linux4sam/meta-atmel/pull/187

File: recipes-bsp/dt-overlay-at91/dt-overlay-at91_git.bb (hunk 1 above)
Comment (noglitch) [reply recipes-bsp/dt-overlay-at91/dt-overlay-at91_git.bb:29]: ok, got it. Thanks for your input @kraj !

**Pull request closed without merging**